packages/components/addon/components/hds/form/masked-input/base.hbs

```diff
@@ -2,33 +2,38 @@
   Copyright (c) HashiCorp, Inc.
   SPDX-License-Identifier: MPL-2.0
 }}
-<div class={{this.classNames}} {{style width=@width}}>
-  {{#if @isMultiline}}
-    <Hds::Form::Textarea::Base
-      class="hds-form-masked-input__control"
-      @value={{@value}}
-      @isInvalid={{@isInvalid}}
-      @height={{@height}}
-      id={{this.id}}
-      ...attributes
-    />
-  {{else}}
-    <Hds::Form::TextInput::Base
-      class="hds-form-masked-input__control"
-      @value={{@value}}
-      @isInvalid={{@isInvalid}}
-      id={{this.id}}
-      ...attributes
-    />
+<div class="hds-form-masked-input__wrapper">
+  <div class={{this.classNames}} {{style width=@width}}>
+    {{#if @isMultiline}}
+      <Hds::Form::Textarea::Base
+        class="hds-form-masked-input__control"
+        @value={{@value}}
+        @isInvalid={{@isInvalid}}
+        @height={{@height}}
+        id={{this.id}}
+        ...attributes
+      />
+    {{else}}
+      <Hds::Form::TextInput::Base
+        class="hds-form-masked-input__control"
+        @value={{@value}}
+        @isInvalid={{@isInvalid}}
+        id={{this.id}}
+        ...attributes
+      />
+    {{/if}}
+    <button
+      class="hds-form-masked-input__toggle-button"
+      type="button"
+      aria-label={{this.ariaLabel}}
+      aria-controls={{this.id}}
+      {{on "click" this.onClickToggle}}
+    >
+      <FlightIcon @name={{if this.isMasked "eye" "eye-off"}} @size="16" @isInlineBlock={{false}} />
+      <span class="sr-only" aria-live="polite">{{this.ariaMessageText}}</span>
+    </button>
+  </div>
+  {{#if @hasCopyButton}}
+    <Hds::Copy::Button @text="Copy masked content" @isIconOnly={{true}} @textToCopy="{{@value}}" />
```

Comment:
Is there any reason why a consumer might want to access this `@text`? Understanding that it's fulfilling the `aria-label` since this is an icon-only variant. I think this is probably good, but if a screen reader user is listening to "Copy masked content" is that sufficient to communicate the actual value that's being copied? (e.g., "Copy Terraform cloud team token"?

I'm probably out of my depth here, but would it be useful to concatenate `@text="Copy {{ariaLabel}}"`? Obviously my Ember string interpolation knowledge is strong 😁

Reply:
Great observation here! Interpolating the field label (be that, for example, Datadog password) is a good idea – assuming the copy is not terrible (for example, Please type your Datadog password). 

However, there are a few challenges. The `ariaLabel` above is from the visibility toggle button and reads 'Show masked content'/'Hide masked content'. Being at the `::Base` level, we don't have access to the actual field label (only available in the `::Form` component). What we can do, perhaps, is allowing the user to customize this copy and make it more relevant to their usecase.

Reply:
> allowing the user to customize this copy and make it more relevant to their usecase

+1

Reply:
That might be helpful, but I understand leaving it since it adds an additional argument. Could be something that is optional in that we expose the argument but provide a fallback (e.g., the text you have in there now) in case nothing is passed to it.

Reply:
yeah, that's what I'm thinking, to allow them to override – will update

Reply:
Added a `copyButtonText` to allow a custom value here